test/parallel/test-buffer-badhex.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+
+var Buffer = require('buffer').Buffer;
+
+// Test hex strings and bad hex strings
+{
+  const buf1 = Buffer.alloc(4);
+  assert.equal(buf1.length, 4);
+  assert.deepStrictEqual(buf1, new Buffer([0, 0, 0, 0]));
+  assert.equal(buf1.write('abcdxx', 0, 'hex'), 2);
+  assert.deepStrictEqual(buf1, new Buffer([0xab, 0xcd, 0x00, 0x00]));
+  assert.equal(buf1.toString('hex'), 'abcd0000');
+  assert.equal(buf1.write('abcdef01', 0, 'hex'), 4);
+  assert.deepStrictEqual(buf1, new Buffer([0xab, 0xcd, 0xef, 0x01]));
+  assert.equal(buf1.toString('hex'), 'abcdef01');
+
+  const buf2 = Buffer.from(buf1.toString('hex'), 'hex');
+  assert.equal(buf1.toString('hex'), buf2.toString('hex'));
+
+  const buf3 = Buffer.alloc(5);
+  assert.equal(buf3.write('abcdxx', 1, 'hex'), 2);
+  assert.equal(buf3.toString('hex'), '00abcd0000');
+
+  const buf4 = Buffer.alloc(4);
+  assert.deepStrictEqual(buf4, new Buffer([0, 0, 0, 0]));
+  assert.equal(buf4.write('xxabcd', 0, 'hex'), 0);
+  assert.deepStrictEqual(buf4, new Buffer([0, 0, 0, 0]));
+  assert.equal(buf4.write('xxab', 1, 'hex'), 0);
```

Comment:
nit: `assert.strictEqual` is generally preferred over `assert.equal`, especially for comparisons with `0`. :)